src/compiler/compile/nodes/shared/Context.ts

```diff
@@ -44,13 +52,55 @@ export function unpack_destructuring(contexts: Context[], node: Node, modifier:
 				const key = property.key as Identifier;
 				const value = property.value;
 
-				used_properties.push(x`"${(key as Identifier).name}"`);
+				used_properties.push(x`"${key.name}"`);
 				if (value.type === 'AssignmentPattern') {
-					unpack_destructuring(contexts, value.left, node => x`${modifier(node)}.${key.name} !== undefined ? ${modifier(node)}.${key.name} : ${value.right}` as Node);
+					const n = contexts.length;
+
+					unpack_destructuring(contexts, value.left, node => {
+						const alternate = JSON.parse(JSON.stringify(value.right));
```

Comment:
why cloning the node?

Reply:
because the node will be altered in `update_reference` and I cannot get it right without cloning it. I noticed that `modifier` will be called several times in the process, if the node changes during the process (as `value.right` is basically a closure to `modifier`), then the conditions for finding reference in `update_reference` will be failed as the node has been replaced for the next call.